Sources/DistributedActors/Cluster/SWIM/SWIMActorShell.swift

```diff
@@ -15,9 +15,517 @@
+        case .alive, .suspect:
+            reachability = .reachable
+        case .unreachable, .dead:
+            reachability = .unreachable
+        }
+
+        guard let uniqueNode = change.member.node.asUniqueNode else {
+            self.log.warning("Unable to emit failureDetectorReachabilityChanged, for event: \(change), since can't represent member as uniqueNode!")
+            return
+        }
+
+        self.clusterRef.tell(.command(.failureDetectorReachabilityChanged(uniqueNode, reachability)))
+    }
+    
+    private func handleGossipPayloadProcessedDirective(_ directive: SWIM.Instance.GossipProcessedDirective) {
+        switch directive {
+        case .applied(let change):
+            self.tryAnnounceMemberReachability(change: change)
+        }
+    }
+    
+    /// We have to handle *every* response, because they adjust the value of the timeouts we'll be using in future probes.
+    private func handleEveryPingRequestResponse(response: SWIM.PingResponse, pinged: SWIMPeer) {
+        // self.tracelog(.receive(pinged: pinged.node), message: "\(response)")
+        let directives = self.swim.onEveryPingRequestResponse(response, pinged: pinged)
+
+        if !directives.isEmpty {
+            fatalError("""
+            Ignored directive from: onEveryPingRequestResponse! \
+            This directive used to be implemented as always returning no directives. \
+            Check your shell implementations if you updated the SWIM library as it seems this has changed. \
+            Directive was: \(directives), swim was: \(self.swim.metadata)
+            """)
+        }
+    }
+    
+    // ==== ------------------------------------------------------------------------------------------------------------
+    // MARK: Distributed functions
+
+    distributed func ping(
+        origin: SWIMActorShell,
+        payload: SWIM.GossipPayload,
+        sequenceNumber: SWIM.SequenceNumber
+    ) async -> SWIM.PingResponse {
+        await withCheckedContinuation { continuation in
+            self.log.trace("Received ping@\(sequenceNumber)", metadata: self.swim.metadata([
+                "swim/ping/origin": "\(origin.id)",
+                "swim/ping/payload": "\(payload)",
+                "swim/ping/seqNr": "\(sequenceNumber)",
+            ]))
+
+            self.swim.onPing(
+                pingOrigin: origin,
+                payload: payload,
+                sequenceNumber: sequenceNumber
+            ).forEach { directive in
+                switch directive {
+                case .gossipProcessed(let gossipDirective):
+                    self.handleGossipPayloadProcessedDirective(gossipDirective)
+
+                case .sendAck(_, let pingedTarget, let incarnation, let payload, let sequenceNumber):
+                    continuation.resume(returning: .ack(target: pingedTarget, incarnation: incarnation, payload: payload, sequenceNumber: sequenceNumber))
+                }
+            }
+        }
+    }
```

Comment:
The continuation is here only because forEach isn't async right?

We can/should use a for loop though then we'd be able to avoid the continuation and just return 👍

Reply:
removed continuation https://github.com/apple/swift-distributed-actors/pull/977/commits/e238478d23e6eb83298e6a721b818783c2d99192